test/unit/client.test.ts

```diff
@@ -269,10 +269,94 @@ describe('backoffConnectRetryHandler', () => {
         status: 401
       } as any
     );
-    expect(client['backoffConnectRetryHandler']({ maxConnectionAttempts: 10 }, error, 1)).toBeFalsy();
+    expect(await client['backoffConnectRetryHandler']({ maxConnectionAttempts: 10 }, error, 1)).toBeFalsy();
   });
+  
+  it('should wait until retryAfter has elapsed (axiosError)', async () => {
+    jest.useFakeTimers();
+
+    const error = new AxiosError(
+      'fake error',
+      'FAKE_ERROR',
+      {
+        url: 'fakeUrl',
+        method: 'get'
+      },
+      {},
+      {
+        status: 429,
+        headers: {
+          'retry-after': '13'
+        }
+      } as any
+    );
+
+    const errorSpy = jest.spyOn(client.logger, 'error');
+    const debugSpy = jest.spyOn(client.logger, 'debug');
+
+    const promise = client['backoffConnectRetryHandler']({ maxConnectionAttempts: 10 }, error, 1);
+
+    await flushPromises();
+    expect(errorSpy).toHaveBeenCalledWith(expect.stringContaining('respecting retry-after header'), expect.anything(), expect.anything());
+    expect(debugSpy).not.toHaveBeenCalled();
+
+    await flushPromises();
+    jest.advanceTimersByTime(10000);
+    await flushPromises();
+    expect(debugSpy).not.toHaveBeenCalled();
+
+    jest.advanceTimersByTime(4000);
+    await flushPromises();
+    expect(debugSpy).toHaveBeenCalled();
+
+    await expect(promise).resolves.toBeTruthy();
+    jest.useRealTimers();
+  });
+  
+  it('should wait until retryAfter has elapsed (xmlhttperror)', async () => {
+    jest.useFakeTimers();
+
+    const error = {
+      response: {
+        status: 429,
+        getResponseHeader: () => '12'
+      }
+    };
+
+    const errorSpy = jest.spyOn(client.logger, 'error');
+    const debugSpy = jest.spyOn(client.logger, 'debug');
+
+    const promise = client['backoffConnectRetryHandler']({ maxConnectionAttempts: 10 }, error, 1);
+
+    await flushPromises();
+    expect(errorSpy).toHaveBeenCalledWith(expect.stringContaining('respecting retry-after header'), expect.anything(), expect.anything());
+    expect(debugSpy).not.toHaveBeenCalled();
+
+    await flushPromises();
+    jest.advanceTimersByTime(10000);
+    await flushPromises();
+    expect(debugSpy).not.toHaveBeenCalled();
+
+    jest.advanceTimersByTime(4000);
```

Comment:
I'm not sure if you were trying to closely match the times in the errors, but I noticed you used `12` in this test, so I suppose this could be `3000` if that's important to you. 🤷‍♂️